test.sh

```diff
@@ -1,32 +1,40 @@
-#!/bin/sh -ex
+#!/bin/sh -e
+
+run() {
+    normal='\e[0m'
+    yellow='\e[33m'
+    printf "${yellow}%s${normal}\n" "$*"
+
+    "$@"
```

Comment:
To clarify, I believe @daurnimator meant to use `$*` instead of the **entire** line, not just the section inside the quotes.